### PR TITLE
feat(time-picker): AM/PM support

### PR DESCRIPTION
**Pull request by Supamiu**
## Description

Add support for AM/PM display inside the time-picker, *enabled by default based on LOCALE_ID in combination with Intl checks*.

Interactions are:
- Going over 12AM switches display to PM, works the other way around and cycles
- Pasting time in a 24h format converts its display to AM/PM if enabled
- Focusing AM/PM part and using keyboard arrows switches from one to the other
- Using "A" or "P" keys (with or without Shift enabled) switches to the appropriate display

**This only affects the display, output model is still using 24h format**

-----

I'm not a big fan of how I ended up detecting AM/PM using Intil but couldn't find a reliable way to find that other than using `format` since `resolvedOptions()` doesn't expose time format based on locale.

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18688/ui-diff)

File: packages/scss/src/components/timepicker/component.scss [hunk 1]
```diff
@@ -31,12 +31,17 @@
...
 
 		.timePicker-fieldset-groupSeparator {
-			text-align: center;
-			position: absolute;
-			inset: 0;
+			// text-align: center;
+			// position: absolute;
+			// top: 0;
+			// bottom: 0;
+			// left: var(--components-timepicker-separator-left);
```
Comment (vvalentin-lucca) [inline packages/scss/src/components/timepicker/component.scss:38]: I have to delete these comments.

File: packages/scss/src/components/timepicker/component.scss [hunk 2]
```diff
@@ -31,12 +31,17 @@
...
 			pointer-events: none;
 			display: grid;
 			place-items: center;
+			text-align: center;
+			width: 1rem;
+			margin: 0 -0.5rem;
```
Comment (vvalentin-lucca) [inline packages/scss/src/components/timepicker/component.scss:44]: And here I add a CSS var.

**Comment by Supamiu** [comment]
closes #2881

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18809/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18887/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19011/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19256/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19259/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19270/ui-diff)

**Review by jeremie-lucca — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2888/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19565/ui-diff)

**Pull request merged**